cmd/operator/internal/controller/clusterkubearchiveconfig_controller.go

```diff
@@ -0,0 +1,128 @@
+	Client client.Client
+	Scheme *runtime.Scheme
+	Mapper meta.RESTMapper
+}
+
+//+kubebuilder:rbac:groups=kubearchive.kubearchive.org,resources=clusterkubearchiveconfigs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=kubearchive.kubearchive.org,resources=clusterkubearchiveconfigs/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=kubearchive.kubearchive.org,resources=clusterkubearchiveconfigs/finalizers,verbs=update
+//+kubebuilder:rbac:groups=core,resources=serviceaccounts,verbs=create;delete;get;list;update;watch
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterroles;roles;rolebindings,verbs=bind;create;delete;escalate;get;list;update;watch
+//+kubebuilder:rbac:groups=sources.knative.dev,resources=apiserversources,verbs=create;delete;get;list;update;watch
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=create;delete;get;list;update;watch
+//+kubebuilder:rbac:groups="",resources=namespaces,verbs=get;list;update;watch
+
+func (r *ClusterKubeArchiveConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+
+	log.Info("Reconciling ClusterKubeArchiveConfig")
+
+	ckaconfig := &kubearchivev1alpha1.ClusterKubeArchiveConfig{}
+	if err := r.Client.Get(ctx, req.NamespacedName, ckaconfig); err != nil {
+		// Ignore not-found errors, since they can't be fixed by an immediate requeue (we need
+		// to wait for a new notification), and we can get them on deleted requests.
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	finalizerName := "kubearchive.org/finalizer"
+
+	if ckaconfig.DeletionTimestamp.IsZero() {
+		// The object is not being deleted, add the finalizer if necessary.
+		if !controllerutil.ContainsFinalizer(ckaconfig, finalizerName) {
+			controllerutil.AddFinalizer(ckaconfig, finalizerName)
+			if err := r.Client.Update(ctx, ckaconfig); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	} else {
+		// The object is being deleted.
+		if controllerutil.ContainsFinalizer(ckaconfig, finalizerName) {
+			// Finalizer is present, clean up filters from ConfigMap and remove Namespace label.
+
+			log.Info("Deleting ClusterKubeArchiveConfig")
+
+			// Clean filters.
+			err := deleteNamespaceFromFilterConfigMap(ctx, r.Client, filters.GlobalNamespace)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+
+			// Remove the finalizer from the list and update it.
+			controllerutil.RemoveFinalizer(ckaconfig, finalizerName)
+			if err := r.Client.Update(ctx, ckaconfig); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+
+		// Stop reconciliation as the resource is being deleted.
+		return ctrl.Result{}, nil
+	}
+
+	yaml, err := convertToYamlString(ckaconfig.Spec.Resources)
+	if err != nil {
+		log.Error(err, "Failed to convert ClusterKubeArchiveConfig resources to YAML")
+		return ctrl.Result{}, err
+	}
+
+	cm, err := reconcileFilterConfigMap(ctx, r.Client, filters.GlobalNamespace, yaml)
```

Comment:
same as above

Reply:
It shouldn't.  We should use the imported value. I will remove the local declaration.

Reply:
Fixed.